### PR TITLE
WB-5

### PR DESCRIPTION
**Pull request by duncte123**
Create the command system and add two commands

### CODE REVIEW

File: src/main/java/io/hypesquad/watsonbot/event/WatsonEventListener.java [hunk 1]
```diff
@@ -17,24 +17,53 @@
...
         final String message = event.getMessage().getContent();
+        //change to constant if it's used multiple times
+        final String prefix = WatsonUtil.getProperty("prefix");
 
-        if (!message.startsWith(WatsonUtil.getProperty("prefix"))) //change to constant if it's used multiple times
+        if (!message.startsWith(prefix)) {
             return;
+        }
+
+        //Handle command
```
Comment (kevzz1994) [inline src/main/java/io/hypesquad/watsonbot/event/WatsonEventListener.java:51]: This can be done much easier. Find a better and more logical way.

**Review by kevzz1994 — CHANGES_REQUESTED** [review]

File: src/main/java/io/hypesquad/watsonbot/commands/AbstractWatsonCommand.java [hunk 2]
```diff
@@ -32,7 +32,7 @@
      *
      * @param args  the arguments from the command
      * @param event the MessageReceivedEvent from D4J
-     * @return true or false depending on some checks that we run
+     * @return ture if checks pass, false otherwise
```
Comment (kevzz1994) [inline src/main/java/io/hypesquad/watsonbot/commands/AbstractWatsonCommand.java]: Quite the typo you have there.

Comment (cascer1) [reply src/main/java/io/hypesquad/watsonbot/commands/AbstractWatsonCommand.java]: fffffff

**Review by cascer1 — DISMISSED** [review]

File: src/main/java/io/hypesquad/watsonbot/commands/WatsonHelpCommand.java [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+    @Override
+    public boolean checkCommand(final String[] args, final MessageReceivedEvent event) {
+        return true;
+    }
+
+    @Override
+    public void executeCommand(final String[] args, final MessageReceivedEvent event) {
+
+        final StringBuilder stringBuilder = new StringBuilder(80).append("```\n");
```
Comment (kevzz1994) [inline src/main/java/io/hypesquad/watsonbot/commands/WatsonHelpCommand.java]: Why ``80``?

**Review by kevzz1994 — CHANGES_REQUESTED** [review]

File: src/main/java/io/hypesquad/watsonbot/commands/WatsonHelpCommand.java [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+
+    @Override
+    public boolean checkCommand(final String[] args, final MessageReceivedEvent event) {
+        return true;
+    }
+
+    @Override
+    public void executeCommand(final String[] args, final MessageReceivedEvent event) {
+
+        final StringBuilder stringBuilder = new StringBuilder(20).append("```\n");
```
Comment (kevzz1994) [inline src/main/java/io/hypesquad/watsonbot/commands/WatsonHelpCommand.java]: ``new StringBuilder(20)``, 20 here is a magic number. Refactor to a constant.

**Review by kevzz1994 — CHANGES_REQUESTED** [review]

File: src/main/java/io/hypesquad/watsonbot/commands/AbstractWatsonCommand.java [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+        }
+
+        AbstractWatsonCommand command = (AbstractWatsonCommand) object;
+
+        return this.commandHelp().equals(command.commandHelp());
+    }
+
+    @Override
+    public int hashCode() {
+        return 1;
```
Comment (kevzz1994) [inline src/main/java/io/hypesquad/watsonbot/commands/AbstractWatsonCommand.java]: Hashcode should be calculated differently.

**Review by kevzz1994 — CHANGES_REQUESTED** [review]

**Review by kevzz1994 — APPROVED** [review]

**Pull request merged**